cv32e40p/env/uvme/cov/uvme_rv32x_hwloop_covg.sv

```
@@ -0,0 +1,780 @@
+    `ifdef MODEL_TECH \
+    option.get_inst_coverage    = 1; \
+    `endif \
+    type_option.merge_instances = 1; \
+    cp_lpstart_``LOOP_IDX : coverpoint (csr_hwloop.lp_start[``LOOP_IDX``]) iff (csr_hwloop.lp_start_wb[``LOOP_IDX``] && csr_hwloop.lp_end_wb[``LOOP_IDX``] && csr_hwloop.lp_count_wb[``LOOP_IDX``]) { \
+      bins lpstart_range_0      = {[32'h0000_03FC : 32'h0000_0004]}; \
+      bins lpstart_range_1      = {[32'h0000_0FFC : 32'h0000_0400]}; \
+      bins lpstart_range_2      = {[32'h0000_FFFC : 32'h0000_1000]}; \
+      illegal_bins  others_grp  = default; // higher range is not covered now due to limited generated codespace (amend if needed) \
+    } \
+    cp_lpend_``LOOP_IDX : coverpoint (csr_hwloop.lp_end[``LOOP_IDX``]) iff (csr_hwloop.lp_start_wb[``LOOP_IDX``] && csr_hwloop.lp_end_wb[``LOOP_IDX``] && csr_hwloop.lp_count_wb[``LOOP_IDX``]) { \
+      bins lpend_range_0        = {[32'h0000_03FC : 32'h0000_0004]}; \
+      bins lpend_range_1        = {[32'h0000_0FFC : 32'h0000_0400]}; \
+      bins lpend_range_2        = {[32'h0000_FFFC : 32'h0000_1000]}; \
+      illegal_bins  others_grp  = default; // higher range is not covered now due to limited generated codespace (amend if needed) \
+    } \
+    cp_lpcount_``LOOP_IDX : coverpoint (csr_hwloop.lp_count[``LOOP_IDX``]) iff (csr_hwloop.lp_start_wb[``LOOP_IDX``] && csr_hwloop.lp_end_wb[``LOOP_IDX``] && csr_hwloop.lp_count_wb[``LOOP_IDX``]) { \
+      // bins lpcount_zero           = {32'h0}; // valid CSR writes to sample should be when lpcount{0/1}.value != 0 \
+      bins lpcount_range_low_1    = {[32'h0000_00FF : 32'h0000_0001]}; // 1 <= x <255 \
+      bins lpcount_range_low_2    = {[32'h0000_1FFF : 32'h0000_0100]}; // 256 <= x < 4K \
+      // bins lpcount_range_middle   = {[32'h00FF_FFFF : 32'h0000_1000]}; // 4K <= x < 16M \
+      // bins lpcount_range_high     = {[32'hFFFF_FFFF : 32'h0100_0000]}; // 16M <= x < 4G \
+      illegal_bins other_range    = default; \
```

illegal here is not correct, other values are allowed also

---

Yes, ti can support upto 32bits but in the tests, we never exercise such ranges. Should i uncommented out lines 104 and 105?

---

Excercise or not is not the question here.
Any cover bin is illegal if we want to raise a fatal error in simulation, i.e., we dont expect such value to be generated/sampled ever.

But if we dont want to record or cover some legal value it should be ignored not illegal.

---

thanks..I missed out that we have  ignore_bins